internal-dns-client/src/multiclient.rs

```diff
@@ -0,0 +1,651 @@
+                1111,
+                0,
+                0,
+            ),
+            SocketAddrV6::new(
+                Ipv6Addr::from_str("ff::02").unwrap(),
+                2222,
+                0,
+                0,
+            ),
+            SocketAddrV6::new(
+                Ipv6Addr::from_str("ff::03").unwrap(),
+                3333,
+                0,
+                0,
+            ),
+        ];
+        let clickhouse_addr = SocketAddrV6::new(
+            Ipv6Addr::from_str("fe::01").unwrap(),
+            4444,
+            0,
+            0,
+        );
+        let crucible_addr = SocketAddrV6::new(
+            Ipv6Addr::from_str("fd::02").unwrap(),
+            5555,
+            0,
+            0,
+        );
+
+        let records = vec![
+            // Three Cockroach services
+            TestServiceRecord::new(
+                AAAA::Zone(Uuid::new_v4()),
+                SRV::Service(ServiceName::Cockroach),
+                cockroach_addrs[0],
+            ),
+            TestServiceRecord::new(
+                AAAA::Zone(Uuid::new_v4()),
+                SRV::Service(ServiceName::Cockroach),
+                cockroach_addrs[1],
+            ),
+            TestServiceRecord::new(
+                AAAA::Zone(Uuid::new_v4()),
+                SRV::Service(ServiceName::Cockroach),
+                cockroach_addrs[2],
+            ),
+            // One Clickhouse service
+            TestServiceRecord::new(
+                AAAA::Zone(Uuid::new_v4()),
+                SRV::Service(ServiceName::Clickhouse),
+                clickhouse_addr,
+            ),
+            // One Backend service
+            TestServiceRecord::new(
+                AAAA::Zone(Uuid::new_v4()),
+                SRV::Backend(BackendName::Crucible, Uuid::new_v4()),
+                crucible_addr,
+            ),
+        ];
+        updater.insert_dns_records(&records).await.unwrap();
+
+        // Look up Cockroach
+        let ip = resolver
+            .lookup_ipv6(SRV::Service(ServiceName::Cockroach))
+            .await
+            .expect("Should have been able to look up IP address");
+        assert!(cockroach_addrs.iter().any(|addr| addr.ip() == &ip));
+
+        // Look up Clickhouse
+        let ip = resolver
+            .lookup_ipv6(SRV::Service(ServiceName::Clickhouse))
+            .await
+            .expect("Should have been able to look up IP address");
+        assert_eq!(&ip, clickhouse_addr.ip());
+
+        // Look up Backend Service
+        let ip = resolver
+            .lookup_ipv6(records[4].srv.clone())
+            .await
+            .expect("Should have been able to look up IP address");
+        assert_eq!(&ip, crucible_addr.ip());
+
+        // If we remove the AAAA records for two of the CRDB services,
+        // only one will remain.
+        updater
+            .dns_records_delete(&vec![DnsRecordKey {
+                name: records[0].aaaa.to_string(),
+            }])
```

Comment:
Just curious about the API here, but both `records[0]` and `records[1]` could be deleted simultaneously, right?

Reply:
Yup; might as well show that. Call updated.